authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -0,0 +1,234 @@
+	Decesion string `json:"accessDecision"`
+}
+
+type RoleAssignment struct {
+	Id               string `json:"Id"`
+	RoleDefinitionId string `json:"RoleDefinitionId"`
+	PrincipalId      string `json:"PrincipalId"`
+	PrincipalType    string `json:"PrincipalType"`
+	Scope            string `json:"Scope"`
+	Condition        string `json:"Condition"`
+	ConditionVersion string `json:"ConditionVersion"`
+	CanDelegate      bool   `json:"CanDelegate"`
+}
+
+type AzureRoleAssignment struct {
+	DelegatedManagedIdentityResourceId string `json:"DelegatedManagedIdentityResourceId"`
+	RoleAssignment
+}
+
+type Permission struct {
+	Actions       []string `json:"actions,omitempty"`
+	NoActions     []string `json:"noactions,omitempty"`
+	DataActions   []string `json:"dataactions,omitempty"`
+	NoDataActions []string `json:"nodataactions,omitempty"`
+}
+
+type Principal struct {
+	Id   string `json:"Id"`
+	Type string `json:"Type"`
+}
+
+type DenyAssignment struct {
+	Id          string `json:"Id"`
+	Name        string `json:"Name"`
+	Description string `json:"Description"`
+	Permission
+	Scope                   string `json:"Scope"`
+	DoNotApplyToChildScopes bool   `json:"DoNotApplyToChildScopes"`
+	principals              []Principal
+	excludeprincipals       []Principal
+	Condition               string `json:"Condition"`
+	ConditionVersion        string `json:"ConditionVersion"`
+}
+type AzureDenyAssignment struct {
+	IsSystemProtected string `json:"IsSystemProtected"`
+	DenyAssignment
+}
+
+type AuthorizationDecesion struct {
+	Decesion            string              `json:"accessDecision"`
+	ActionId            string              `json:"actionId"`
+	IsDataAction        bool                `json:"isDataAction"`
+	AzureRoleAssignment AzureRoleAssignment `json:"roleAssignment"`
+	AzureDenyAssignment AzureDenyAssignment `json:"denyAssignment"`
+	TimeToLiveInMs      int                 `json:"timeToLiveInMs"`
+}
+
+// this will be removed once caching is implemented
+func getUserId(userName string) string {
+	switch userName {
+	case "krdhruva@microsoft.com":
+		return "63e8a863-9ae9-4f3c-b0b7-fd9df05c712e"
+	case "test@KDOrg.onmicrosoft.com":
+		return "62103f2e-051d-48cc-af47-b1ff3deec630"
+	default:
+		return "62103f2e-051d-48cc-af47-b1ff3deec630"
+	}
+}
+
+func getScope(resourceId string, attr *authzv1.ResourceAttributes) string {
+	if attr != nil && attr.Namespace != "" {
+		return resourceId + "/namespace/" + attr.Namespace
+	}
+	return resourceId
+}
+
+func IsValidUUID(u string) bool {
+	_, err := uuid.Parse(u)
+	return err == nil
+}
+
+func getSecGroups(groups []string) []string {
+	var finalGroups []string
+	for _, element := range groups {
+		if IsValidUUID(element) {
+			finalGroups = append(finalGroups, element)
+		}
+	}
+	return finalGroups
+}
+
+func getActionName(verb string) string {
+	switch verb {
+	case "get":
+		fallthrough
+	case "list":
+		fallthrough
+	case "watch":
+		return "read"
+	case "create":
+		return "action"
```

Comment:
I know Azure API is not very RESTFul. but can we be conforming to Azure (poor) standard please

Reply:
updated function with other verbs